### ISSUE TITLE
[libbeat] Kafka output doesn't respect retry backoff settings when host is unreachable

### ISSUE THREAD

**Issue by faec** [elastic/beats#22437]
- Version: Filebeat 7.9.3
- Operating System: Mac OS 10.14.6

(This seems to affect many versions and platforms, but explicitly confirmed on the above.)

To reproduce, run Filebeat on any input source with the following output configuration:

```yml
output.kafka:
    # initial brokers for reading cluster metadata
    hosts: ["unreachableHost:9092"]
    version: 2.1
    backoff:
      init: 3s
      max: 30s
  
    topic: 'foo'
```

This produces logs of constant failures with no apparent pause between retries:

```
...
2020-11-04T13:15:09.838-0500    INFO    [publisher]     pipeline/retry.go:219   retryer: send unwait signal to consumer
2020-11-04T13:15:09.838-0500    INFO    [publisher]     pipeline/retry.go:223     done
2020-11-04T13:15:09.853-0500    INFO    [publisher]     pipeline/retry.go:219   retryer: send unwait signal to consumer
2020-11-04T13:15:09.853-0500    INFO    [publisher]     pipeline/retry.go:223     done
2020-11-04T13:15:09.869-0500    INFO    [publisher]     pipeline/retry.go:219   retryer: send unwait signal to consumer
2020-11-04T13:15:09.869-0500    INFO    [publisher]     pipeline/retry.go:223     done
...
```

**Comment by Djeezus** [comment]
Hi, I confirm this bug too with "filebeat-7.10.0-1.x86_64"; got reported due to high (99%) cpu usage due to a network outage...
Is there an ETA for implementation of "retryer.maxbackoff" in an official release ?  https://github.com/elastic/beats/pull/22135 seems to be missing 1 review for merging to take place ...

**Comment by ZHumphries** [comment]
Can confirm this is also a problem with winlogbeat 7.10. Be careful with this on virtual infrastructure, it has the potential to bring the entire network to a grinding halt.

**Comment by faec** [comment]
Fixed as of 7.11
